docs/DevGuide/TensorExpressions.md

```diff
@@ -0,0 +1,411 @@
+\cond NEVER
+Distributed under the MIT License.
+See LICENSE.txt for details.
+\endcond
+# Writing tensor equations with TensorExpressions {#writing_tensorexpressions}
+
+\tableofcontents
+
+SpECTRE's `TensorExpression`s interface allows you to write tensor equations in
+SpECTRE in C++ with syntax that resembles tensor index notation. To use it,
+simply add this include to the top of your file:
+```
+#include "DataStructures/Tensor/Tensor.hpp"
+```
+The following guide assumes a basic understanding of the `Tensor` class and
+\ref tnsr "tnsr" type aliases.
+
+# Syntax {#te_syntax}
+`TensorExpression`s are arithmetic expressions of `Tensor`s that can be
+evaluated using `tenex::evaluate`. Terms used in the expression may be `Tensor`s
+or numbers (see [supported types](#te_data_type_support)).
+
+As a simple example of how `TensorExpression`s are used, if you would like to
+raise the index of some `Tensor` `R` with some spacetime metric `Tensor` `g`,
+i.e. \f$R^c{}_b = R_{ab} g^{ac}\f$, you can compute this with
+`TensorExpression`s by doing:
+```
+auto R_up = tenex::evaluate<ti::C, ti::b>(R(ti::a, ti::b) * g(ti::A, ti::C));
+```
+
+where `R_up`, `R`, and `g` are rank 2 spacetime `Tensor`s and the `ti::*`
+variables are `TensorIndex`s. Here, the argument to
+\ref tenex::evaluate "evaluate" is the RHS tensor expression to compute, `R_up`
+is the result LHS `Tensor`, and the template arguments to
+\ref tenex::evaluate "evaluate" are the LHS `Tensor`'s indices. The LHS symmetry
+will be deduced from the RHS tensors' symmetries and order of operations.
+
+Alternatively, if you already have a LHS `Tensor` variable, you can pass it into
+the following \ref tenex::evaluate "evaluate" overload, where the LHS `Tensor`
+provided will be assigned to the result of the RHS expression:
+
+```
+tenex::evaluate<ti::C, ti::b>(
+    make_not_null(&R_up), R(ti::a, ti::b) * g(ti::A, ti::C));
+```
+
+One advantage of this overload is that it uses the symmetry and index types
+(spatial/spacetime) of the provided LHS tensor instead of deducing them from the
+RHS expression, so this enables you to specify the LHS index structure in cases
+where the previous \ref tenex::evaluate "evaluate" overload does not deduce the
+one you want. Note that the LHS tensor does not need to be previously sized
+unless the data type is a Blaze vector type (e.g. `DataVector`) *and* the RHS
+expression contains no `Tensor` terms (see [example](#te_assigning_to_a_number)
+where sizing is necessary). This overload is also used to
+[assign subsets of tensor components](#te_assigning_subsets_of_components).
+
+## Tensor indices (TensorIndexs) {#te_tensor_indices}
+
+`TensorIndex`s represent generic tensor indices and are supplied as
+comma-separated lists in two places: (1) in parentheses for each tensor in the
+RHS expression and (2) in the template parameters of
```

Comment:
Can this be an actual numbered list like the bullets below? Draws the eyes to it more.